mycore-mets/src/main/java/org/mycore/mets/resource/MCRAltoHighlightResource.java

```diff
@@ -170,7 +170,17 @@ private JsonArray buildQueryResponse(Map<String, Map<String, List<String>>> high
             JsonArray positions = new JsonArray();
             int wordCount = StringUtils.countMatches(altoContent, "<em>");
             for (int i = wordIndex; i < wordIndex + wordCount; i++) {
-                positions.add(buildPositionObject(altoWords.get(i)));
+                String altoWord = altoWords.get(i);
+
+                if(!altoWord.equals("")){
+                    JsonObject positionObject = buildPositionObject(altoWord);
+
+                    if(!positionObject.entrySet().isEmpty()){
+                        positions.add(positionObject);
+                    }
+                } else if(wordCount < altoWords.size()){
```

Comment:
The if-part looks O.K. for me - I used a similiar solution to temporarily fix the problem  in my own application
Can you please explain, what you are doing in the else-Part?

Reply:
As you can see in the Jira ticket, the fix will address the issue with Solr 8.8 and higher. For responses from Solr with "hl.method":"unified" you will get "alto_words" : "["", "<em>some alto</em"]". In this case the wordCount is not correct and you have to increase it to get the alto. Maybe there will be a better solution, this is what came first to my mind to fix the issue. I don't know if will cause errors for some other unkown cases.